lib/arrays/implementations/map_array.ex

```diff
@@ -6,20 +6,10 @@ defmodule Arrays.Implementations.MapArray do
   """
   defstruct contents: %{}, default: nil
 
-  def empty(options) do
-    default = Keyword.get(options, :default, nil)
-    size = Keyword.get(options, :size, 0)
-    %MapArray{contents: construct(default, size), default: default}
-  end
-
-  defp construct(_default, 0), do: %{}
-
-  defp construct(default, size) do
-    Enum.into(0..(size - 1), %{}, &{&1, default})
-  end
-
   use FunLand.Mappable
 
+  defdelegate empty(option \\ []), to: Arrays.Protocol.Arrays.Implementations.MapArray
```

Comment:
We should use reflection here rather than the hard-coded protocol implementation module name, because AFAIK that is an implementation detail which might be changed in future versions of Elixir.

Hmm.... there is however no clean way to go from a module name to a protocol module (only from a _struct_ or other data type to a protocol module).

Maybe it's better to do it the other way around: Let the protocol implementation module delegate to this explicit `empty` in the `Arrays.Implementations.MapArray` module.
What do you think?